versions/1.1/SPEC.md

```diff
@@ -0,0 +1,5846 @@
+
+The following guidelines should be followed when using hints:
+
+* A hint should never be required.
+* Less is more. Before adding a new hint key, ask yourself "do I really need another hint?", or "is there a better way to specify the behavior I require". If there is, then adding a hint is likely not the right way to achieve your aims.
+* Complexity is killer. By allowing any arbitrary keys, it is possible that the `runtime` section can get quite unruly and complex. This should be discouraged, and instead a pattern of simplicity should be stressed.
+* Sharing is caring. People tend to look for similar behavior between different execution engines. It is strongly encouraged that execution engines agree on common names and accepted values for hints that describe common usage patterns. A good example of hints that have conventions attached to them is cloud provider specific details:
+    ```wdl
+    task foo {
+      .... 
+      runtime {
+        gcp: {
+        ...
+        }
+        aws: {
+        ...
+        }
+        azure: {
+          ...
+        }
+        alibaba: {
+          ...
+        }
+      }
+    }
+    ```
+* Use objects to avoid collisions. If there are specific hints that are unlikely to ever be shared between execution engines, it is good practice to encapsulate these within their own execution engine-specific hints object:
+    ```wdl
+    task foo {
+      .... 
+      runtime {
+        cromwell: {
+          # cromwell specific 
+          ...
+        }
+        miniwdl: {
+          # miniwdl specific
+          ...
+        }
+      }
+    }
+    ```
+
+### Metadata Sections
+
+```txt
+$meta_kv = $identifier $ws* '=' $ws* $meta_value
+$meta_value = $string | $number | $boolean | 'null' | $meta_object | $meta_array
+$meta_object = '{}' | '{' $meta_kv (, $meta_kv)* '}'
+$meta_array = '[]' |  '[' $meta_value (, $meta_value)* ']'
+```
+
+There are two purely optional sections that can be used to store metadata with the task: `meta` and `parameter_meta`. These sections are designed to contain metadata that is only of interest to human readers. The engine can ignore these sections with no loss of correctness. The extra information can be used, for example, to generate a user interface. Any attributes that may influence execution behavior should go in the `runtime` section.
+
+Both of these sections can contain key/value pairs. Metadata values are different than in `runtime` and other sections:
+
+* Only string, numeric, and boolean primitives are allowed.
+* Only array and "meta object" compound values are allowed.
+* The special value `null` is allowed for undefined attributes.
+* Expressions are not allowed.
+
+A meta object is similar to a regular WDL object, except:
+* The `object` keyword is not required.
+* Its values must conform to the same metadata rules defined above.
+
+Note that, unlike the WDL `Object` type, metadata objects are not deprecated and will continue to be supported in future versions.
```

Comment:
If you discuss these as `structs` you don't have to include this confusing sentence.